### PR TITLE
Check forbidden branches in pre-commit hook

### PR DESCRIPTION
**Pull request by Belco90**
## Changes

Since I have an admin role, I can push to `main` without a problem. This is leading me to push without realizing I'm working in the default branch.

I'm including this small pre-commit hook to prevent committing to `main` branch so I can move changes to the corresponding branch to follow the PR workflow. It can be bypassed with `-n` option if I need to push directly to `main` for whatever reason.

## Context

 There wasn't specific ticket for this.

### CODE REVIEW

**Comment by Belco90** [comment]
The proof this works:
  
CleanShot 2021-07-17 at 11 52 00@2x

File: .husky/forbidden-branches.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# shellcheck disable=SC2006
+BRANCH=`git rev-parse --abbrev-ref HEAD`
```
Comment (HonkingGoose) [inline .husky/forbidden-branches.sh:2]:
```suggestion
BRANCH=$(git rev-parse --abbrev-ref HEAD)
```

I ran the shell script through the shellcheck program, it suggests we use this syntax. I guess there's a reason you use the backtick syntax?

**Review by HonkingGoose — COMMENTED** [review]
I like this idea a lot! 😄

One question though:

File: .husky/forbidden-branches.sh (hunk 1 above)
Comment (Belco90) [reply .husky/forbidden-branches.sh:2]: You are right! My IDE was warning me about it, but when I told it to fix it, it wasn't changed 🤔 

Let me fix this.

Comment (Belco90) [reply .husky/forbidden-branches.sh]: Done! I've moved the script outside husky path since files there are supposed to match a git hook.

**Comment by HonkingGoose** [comment]
I checked out this PR branch locally, and I tried to commit on top of this branch, and Husky complains:

```bash
.husky/pre-commit: line 8: .: forbidden-branches.sh: file not found
husky - pre-commit hook exited with code 1 (error)
```

I'm using Git for Windows, you use ZSH, so maybe there's an incompatibility here?

**Comment by Belco90** [comment]
> I checked out this PR branch locally, and I tried to commit on top of this branch, and Husky complains:
> 
> ```shell
> .husky/pre-commit: line 8: .: forbidden-branches.sh: file not found
> husky - pre-commit hook exited with code 1 (error)
> ```
> 
> I'm using Git for Windows, you use ZSH, so maybe there's an incompatibility here?

I've moved the script within pre-commit file, can you give it a try again? Thanks for double-checking!

**Comment by HonkingGoose** [comment]
> I've moved the script within pre-commit file, can you give it a try again? Thanks for double-checking!

That seems to have done the trick, I can commit on this PR branch now.
Not sure if the script will actually work to prevent me committing to `main`, I think the only way to find out is to merge this PR... 😄

**Comment by Belco90** [comment]
> Not sure if the script will actually work to prevent me committing to `main`, I think the only way to find out is to merge this PR... 😄

I checked it before pushing, it's still working!

File: .husky/pre-commit [hunk 2]
```diff
@@ -5,3 +5,14 @@
 [ -n "$CI" ] && exit 0
 
 yarn lint-staged
+
+# Prevent commit if default branch
```
Comment (HonkingGoose) [inline .husky/pre-commit:9]:
```suggestion
# Prevent commit if on main branch
```

I think this comment is more accurate?

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

**Review by HonkingGoose — APPROVED** [review]

**Comment by HonkingGoose** [comment]
I confirm this works properly for me now as well. I cannot commit to `main` without Husky warning me about it.

**Pull request merged**